common/include/pcl/common/point_tests.h

```diff
@@ -101,4 +101,20 @@ namespace pcl
   {
     return (std::isfinite (n.normal_x) && std::isfinite (n.normal_y) && std::isfinite (n.normal_z));
   }
+
+  template <class PointT, traits::HasXY<PointT> = true> inline bool
```

Comment:
Don't we need a "catch all" version for when the point type does not have XY fields?

Reply:
Is `HasXY<PointT>` fails, it gives an error message. Why would a "catch all" be required?

Reply:
Let's say I have a templated algorithm class that supports point types both without and with normals. At some point it needs to check whether an input point is valid, i.e. has finite coordinates. In the case of points with normals, this check should also involve normal coordinates. What I would like to be able to write in such class i

```cpp
if(!isXYZFinite(pt) || !isNormalFinite(pt))
  continue;
```

and not worry that the second component of the expression is not defined for normal-less points.

Reply:
I see. So basically a "`return true` if the required fields do not exist for this point type" function?

Reply:
Yes. This is sort of how it is right now with `isFinite()` function.